### PR TITLE
fix references and examples

### PR DESCRIPTION
**Pull request by zoellner**
a couple of references were outdated due to the removal of sections. some examples were outdated and didn't match the specification.

***
<a href="https://pr-preview.s3.amazonaws.com/zoellner/did-spec/pull/48.html" title="Last updated on Feb 6, 2018, 6:44 PM GMT (eda9a2d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/did-spec/48/4e6c77c...zoellner:eda9a2d.html" title="Last updated on Feb 6, 2018, 6:44 PM GMT (eda9a2d)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1046,12 +1046,16 @@ <h2>Authentication</h2>
   "id": "did:example:123456789abcdefghi",
   ...
   "authentication": [{
-    // this key can be used to authenticate as DID ...9938
+    // this key can be used to authenticate as DID ...fghi
+    "id": "did:example:123456789abcdefghi#keys-1"
```
Comment (msporny) [inline index.html]: This is a link to the key, which is wrong. The object being edited is an authentication suite, not a key object. The authentication suite doesn't have an "id".

File: index.html [hunk 2]
```diff
@@ -1046,12 +1046,16 @@ <h2>Authentication</h2>
...
     "type": "RsaSignatureAuthentication2018",
+    "owner": "did:example:123456789abcdefghi"
```
Comment (msporny) [inline index.html]: authentication suites do not have owners.

File: index.html [hunk 3]
```diff
@@ -1046,12 +1046,16 @@ <h2>Authentication</h2>
...
     "publicKey": "did:example:123456789abcdefghi#keys-1"
   }, {
-    // this key can be used to authenticate as DID ...9938
+    // this key can be used to authenticate as DID ...fghi
+    "id": "did:example:123456789abcdefghi#bio-1"
```
Comment (msporny) [inline index.html]: authentication suites do not have ids

File: index.html [hunk 4]
```diff
@@ -1046,12 +1046,16 @@ <h2>Authentication</h2>
...
     "type": "PseudonymousBiometricAuthentication2018",
+    "owner": "did:example:123456789abcdefghi"
```
Comment (msporny) [inline index.html]: authentication suites do not have owners.

File: index.html [hunk 5]
```diff
@@ -1384,7 +1388,7 @@ <h2>Updated (Optional)</h2>
 
   <li>
 The value of this key MUST follow the formatting rules (3, 4, 5)
-from section 6.7.
+from section 4.5.
```
Comment (msporny) [inline index.html]: While this is accurate, it's just going to get messed up again unless we use ReSpec's <a href="#frag-id"></a> format. We should update to use that pattern everywhere.

File: index.html [hunk 6]
```diff
@@ -1592,7 +1596,7 @@ <h1>Security Considerations</h1>
       As the root identifier records for decentralized identifiers, DIDs and DID Documents are a vital component of decentralized identity management. They are also the foundational building blocks of DPKI (decentralized public key infrastructure) as an augmentation to conventional X.509 certificates.
 
 
-      As such, DIDs are designed to operate under the general Internet threat model used by many IETF standards. We assume uncompromised endpoints, but allow messages to be read or corrupted on the network. Protecting against an attack when a system is compromised requires external key-signing hardware. See also section 6.5 regarding key revocation and recovery.
+      As such, DIDs are designed to operate under the general Internet threat model used by many IETF standards. We assume uncompromised endpoints, but allow messages to be read or corrupted on the network. Protecting against an attack when a system is compromised requires external key-signing hardware. See also section <a href="#key-revocation-and-recover"></a> regarding key revocation and recovery.
```
Comment (msporny) [inline index.html]: Yes, this form is correct... but the frag id is wrong... it should be: "#key-revocation-and-recovery"

File: index.html [hunk 7]
```diff
@@ -2147,16 +2151,20 @@ <h1>Real World Example</h1>
   }],
 
   "authentication": [{
-    // this key can be used to authenticate as DID ...9938
+    // this key can be used to authenticate as DID ...fghi
+    "id": "did:example:123456789abcdefghi#keys-1"
```
Comment (msporny) [inline index.html]: Same issue w/ the fixes here... the committer assumed these were key objects, which they are not. They are authentication suite objects. This is probably also a lesson for the CG, that developers will just assume that key information goes in here... I expect many libraries will throw an error, but we'll probably have to include spec text that ensures that they throw on stuff that isn't properly formed.

File: index.html (hunk 6 above)
Comment (zoellner) [reply index.html]: updated to use correct frag id. updated existing links to wrong frag id as well

File: index.html (hunk 5 above)
Comment (zoellner) [reply index.html]: updated to use ReSpec

File: index.html (hunk 7 above)
Comment (zoellner) [reply index.html]: There must be a misunderstanding on my side then about rules 2 and 3 under 4.3 Authentication which states: 
> 2. The value of the authentication property should be an array of proof mechanisms.
> 3. Each proof mechanism must include id, type, and owner properties.

The example in the currently commented out section "Example Delegate-Managed DID Document" has this format as well.

Can you clarify?

Comment (msporny) [reply index.html]: >  id, type, and owner properties

Ah, you have spotted two bugs in the spec. Thank you! Neither id or owner is required. This is old spec text from when we used to put key information in "authentication".

Comment (zoellner) [reply index.html]: I've removed the id and owner properties. I leave it up to you to fix the spec since I want to keep this PR as editorial.

**Comment by zoellner** [comment]
IPR release is signed and all requested changes are addressed
Thank you for the help @msporny

**Comment by msporny** [comment]
LGTM, editorial, squashed and merged to keep history cleaner. Thank you @zoellner!

**Pull request merged**